src/package/packageVersionCreate.ts

```diff
@@ -166,7 +166,7 @@ export class PackageVersionCreate {
     const buildNumber = versionNumber.build;
 
     // use the dependency.branch if present otherwise use the branch of the version being created
-    const branch = dependency.branch === '' ? this.options.branch : dependency.branch;
+    const branch = dependency.branch === undefined ? this.options.branch : dependency.branch;
```

Comment:
@David-Polehonski - let's change lines 169-170 to this equivalent and less verbose code:

```javascript
const branch = dependency.branch ?? this.options.branch;
const branchString = branch ? `'${branch}'` : 'null';
```

Reply:
@shetzel I've made the suggested modification; I agree it's less verbose but it's not quite equivalent as the ?? operator will treat `null` and `undefined` in the same way. I'm not certain what the expected behaviour would be if you specify your dependency with `branch: null` in the project definition.

Reply:
A packaging team member has confirmed that `null` and `undefined` should be equivalent.  I'll merge when tests are passing.  Thanks for the contributions @David-Polehonski and @avesolovksyy